efficientnetv2/main_tf2.py

```diff
@@ -117,190 +91,341 @@ def test_step(self, data):
+    return self._total_steps
+
+  @staticmethod
+  def get_optimizer_tf2(learning_rate: float,
+      optimizer_name: str = 'rmsprop',
+      decay: float=0.9,
+      epsilon: float=0.001,
+      momentum : float=0.9) -> Any:
+
+    """A helper function to build tf2 optimizer for training.
+
+    Args:
+      optimizer_name: string, optimizer for training example (Adam).
+      decay: float. decay rate.
+      epsilon: float. epsilon for the optimizer.
+      momentum. float. momentum used in the optimizer.
+
+    Raises:
+      When passed unknown optimizer raises unknown optimizer.
+
+    """
+
+    if optimizer_name == 'sgd':
+      logging.info('Using SGD optimizer')
+      optimizer = tf.keras.optimizers.SGD(learning_rate=learning_rate)
+    elif optimizer_name == 'momentum':
+      logging.info('Using Momentum optimizer')
+      optimizer = tf.keras.optimizers.SGD(
+          learning_rate=learning_rate, momentum=momentum)
+    elif optimizer_name == 'rmsprop':
+      logging.info('Using RMSProp optimizer')
+      optimizer = tf.keras.optimizers.RMSprop(learning_rate, decay, momentum,
+                                              epsilon)
+    elif optimizer_name == 'adam':
+      logging.info('Using Adam optimizer')
+      optimizer = tf.keras.optimizers.Adam(learning_rate)
     else:
-      ds_strategy = tf.distribute.MirroredStrategy(['CPU:0'])
-
-  with ds_strategy.scope():
-    train_split = config.train.split or 'train'
-    eval_split = config.eval.split or 'eval'
-    num_train_images = config.data.splits[train_split].num_images
-    num_eval_images = config.data.splits[eval_split].num_images
-
-    train_size = config.train.isize
-    eval_size = config.eval.isize
-    if train_size <= 16.:
-      train_size = int(eval_size * train_size) // 16 * 16
-
-    image_dtype = None
-    if config.runtime.mixed_precision:
-      image_dtype = 'bfloat16' if strategy == 'tpu' else 'float16'
-      precision = 'mixed_bfloat16' if strategy == 'tpu' else 'mixed_float16'
-      policy = tf.keras.mixed_precision.Policy(precision)
-      tf.keras.mixed_precision.set_global_policy(policy)
+      raise Exception('Unknown optimizer: %s' % optimizer_name)
+
+    return optimizer
+
+  def setup(self) -> None:
+    """A helper function for setting up model and training setup.
+    setup number of train and eval images, total steps, steps_per_epoch,
+    set the precision of the training, get the efficientnetv2 model
+    instance.
+    """
+    logging.info('Setting up training regime for efficientnetv2')
+
+    self.train_split = self.config.train.split or 'train'
+    self.eval_split = self.config.eval.split or 'eval'
+    self.num_train_images = self.config.data.splits[self.train_split].num_images
+    self.num_eval_images = self.config.data.splits[self.eval_split].num_images
+    self._steps_per_epoch = self.num_train_images // self.config.train.batch_size
+    self._total_steps = self._steps_per_epoch * self.config.train.epochs
```

Comment:
@mingxingtan 
total_steps = max(config.train.min_steps, steps_per_epoch*config.train.epochs)
as in tf1 script to use min_steps